docs/v1/api/slideshow.md

```diff
@@ -0,0 +1,345 @@
+
+## Creating a Slideshow
+
+You can create a slideshow using the `createSlideshow` method of the `Presence` class:
+
+```typescript
+const slideshow = presence.createSlideshow()
+```
+
+## Methods
+
+### addSlide
+
+<!-- eslint-skip -->
+
+```typescript
+addSlide(id: string, data: PresenceData, interval: number): SlideshowSlide;
+```
+
+Adds a slide to the queue. If a slide already exists with the given ID, it will be updated with the new values.
+
+#### Parameters
+
+- `id`: The slide ID (a unique identifier for the slide)
+- `data`: The slide presence data (what will be displayed for this slide)
+- `interval`: Interval until the next slide (in milliseconds, minimum 5000ms)
+
+#### Returns
+
+- A `SlideshowSlide` instance
+
+#### Example
+
+```typescript
+slideshow.addSlide('homepage', {
+  details: 'Browsing the homepage',
+  state: 'example.com',
+  largeImageKey: 'https://example.com/logo.png'
+}, 5000) // 5 seconds
+```
+
+### deleteSlide
+
+<!-- eslint-skip -->
+
+```typescript
+deleteSlide(id: string): void;
+```
+
+Deletes a slide from the queue.
+
+#### Parameters
+
+- `id`: The slide ID
+
+#### Example
+
+```typescript
+slideshow.deleteSlide('homepage')
+```
+
+### deleteAllSlides
+
+<!-- eslint-skip -->
+
+```typescript
+deleteAllSlides(): void;
+```
+
+Clears the queue of all slides.
+
+#### Example
+
+```typescript
+slideshow.deleteAllSlides()
+```
+
+### updateSlide
+
+<!-- eslint-skip -->
+
+```typescript
+updateSlide(id: string, data?: PresenceData, interval?: number): SlideshowSlide;
+```
+
+Updates a slide already in the queue. Passing `null` or `undefined` will keep the old value.
+
+#### Parameters
+
+- `id`: The slide ID
+- `data` (optional): The slide presence data
+- `interval` (optional): Interval until the next slide (in milliseconds)
+
+#### Returns
+
+- A `SlideshowSlide` instance
+
+#### Example
+
+```typescript
+slideshow.updateSlide('homepage', {
+  details: 'Browsing the updated homepage',
+  state: 'example.com',
+  largeImageKey: 'https://example.com/logo.png'
+})
+```
+
+### hasSlide
+
+<!-- eslint-skip -->
+
+```typescript
+hasSlide(id: string): boolean;
+```
+
+Returns whether a slide exists in the queue.
+
+#### Parameters
+
+- `id`: The slide ID
+
+#### Returns
+
+- `true` if the slide exists, `false` otherwise
+
+#### Example
+
+```typescript
+if (slideshow.hasSlide('homepage')) {
+  // Slide exists
+}
+```
+
+### getSlides
+
+<!-- eslint-skip -->
+
+```typescript
+getSlides(): SlideshowSlide[];
+```
+
+Returns all slides in the slideshow.
+
+#### Returns
+
+- An array of `SlideshowSlide` instances
+
+#### Example
+
+```typescript
+const slides = slideshow.getSlides()
+
+// You can then modify all slides at once
+for (const slide of slides) {
+  const data = slide.data
+  data.buttons = [
+    {
+      label: 'Visit Website',
+      url: document.URL
```

Comment:
Most of the Activities in the repository use `document.location.href` instead of `document.URL`

Reply:
@Bas950 thoughts on that? document.URL seems more clear for beginners imo but I'd be using document.location.href a lot personally

Reply:
document.location.href

Reply:
Updated